Web/Scotty/Internal/Types.hs

```diff
@@ -104,6 +120,10 @@ instance ScottyError e => ScottyError (ActionError e) where
 
 type ErrorHandler e m = Maybe (e -> ActionT e m ())
 
+data ScottyException = RequestException BS.ByteString Status deriving (Show, Typeable)
+
+instance Exception ScottyException
```

Comment:
Piggybacking on GHCs internal exceptions channel doesn't sit well with me. I'd prefer we use an explicit error channel that we controlled, such as `ExceptT ScottyException IO` or something like that. However, you have a unit test indicating that the server operates as intended. I think it's fine.